src/python/WMCore/ReqMgr/CherryPyThreads/BuildParentLock.py

```diff
@@ -0,0 +1,102 @@
+        self.dbs = DBS3Reader(config.dbs_url)
+        self.datasetToParentMap = {}
+
+
+    def setConcurrentTasks(self, config):
+        """
+        sets the list of functions which
+        """
+        self.concurrentTasks = [{'func': self.fetchIncludeParentsRequests,
+                                 'duration': config.updateParentsInterval}]
+
+    def fetchIncludeParentsRequests(self, config):
+        """
+        Fetch active requests from the "requestsincludeparents" couch view that
+        have IncludeParents=True, find parents of each dataset and send to
+        reqmgr2 auxiliary database.
+        """
+        # use this boolean to signal whether there were datasets that failed
+        # to get their parentage resolved
+        incompleteParentage = False
+        # use this boolean to signal if new parent datasets need to be locked
+        auxDbUpdateRequired = False
+
+        setDsets = set()
+        setParents = set()
+        dictParents = {}
+
+        self.logger.info("Executing parent lock cherrypy thread")
+
+        # query couch view to find datasets for workflows requiring parent datasets
+        # only returning requests with the statuses in filterKeys
+        reqDB = RequestDBReader(config.reqmgrdb_url)
+        filterKeys = ['assignment-approved', 'assigned', 'staging', 'staged', 'acquired',
+                      'running-open', 'running-closed', 'force-complete', 'completed']
+        results = reqDB._getCouchView("requestsincludeparents", {}, filterKeys)
+        for row in results["rows"]:
+            dataset = row["value"]
+            setDsets.add(dataset)
+
+        # check to see if any changes have been made
+        # this will be true where new datasets are added and old datasets are no longer needed
+        if setDsets ^ set(self.datasetToParentMap.keys()):
+            auxDbUpdateRequired = True
+
+        self.logger.info("Found %d unique datasets requiring the parent dataset", len(setDsets))
+        if auxDbUpdateRequired:
+            self.logger.info("Found new parent dataset locks to update.")
+            # reset dataset to parent dataset map
+            self.datasetToParentMap = {}
+            # look up parent datasets via DBS
+            for dset in setDsets:
+                try:
+                    res = self.dbs.listDatasetParents(dset)
+                except Exception as exc:
+                    self.logger.warning("Failed to resolve parentage for: %s. Error: %s", dset, str(exc))
+                    incompleteParentage = True
+                    continue
+                self.logger.info("Resolved parentage for: %s", res)
+                if res:
+                    setParents.add(res[0]['parent_dataset'])
+                    self.datasetToParentMap[dset] = res[0]['parent_dataset']
+
+            if not incompleteParentage:
+                dictParents['parentlocks'] = list(setParents)
+                self.reqmgrAux.updateParentLocks(dictParents)
```

Comment:
Shouldn't it be something like:
```
if self.reqmgrAux.updateParentLocks(dictParents):
    self.logger.info("Parentage lookup complete and auxiliary database updated.")
else:
    self.logger.warning("Failed to update parentage document. Using stale data until the next cycle.")
```
?

That function returns a boolean to say whether it succeeded or not.